src/program-structure-and-compilation.rst

```diff
@@ -138,6 +138,9 @@ The :t:`function` in scope of a :t:`binary crate`'s :t:`crate root module` under
 the :t:`name` ``main`` with a :t:`main function signature` is the :t:`binary
 crate`'s :t:`program entry point`.
 
+:dp:`fls_d9nn4yuiw1ja`
+A :t:`library crate` is a :t:`crate` that is not a :t:`binary crate`.
```

Comment:
This is not correct, as a proc macro crate is also not a binary crate. I'd specify that it's a crate with the type of `lib`, `rlib`, `staticlib`, `dylib` or `cdylib`.

Reply:
But that definition says so, a proc-macro crate is not a binary crate so it fits the definition of a library crate. Anyways you are right that we ought to be explicit here by listing them.